apps/mobile/src/providers/auth-provider/use-auth-actions.ts

```diff
@@ -22,6 +27,43 @@ const useAuthActions = (): UseAuthActions => {
     enabled: false,
   });
 
+  const { refetch: refetchUserInfo } = userInfoQuery;
+
+  useEffect(() => {
+    const initState = async () => {
+      try {
+        const tokens = await getUserCredentials();
+
+        if (tokens !== null) {
+          persistUserToken(tokens.accessToken);
+          const { data, failureCount } = await refetchUserInfo();
+
+          if (failureCount >= 3) {
+            dispatch({ type: 'SIGN_OUT' });
+            return;
+          }
+
+          if (!data || !data.data) {
```

Comment:
Optional chaining works here?

```suggestion
          if (!data?.data) {
```